src/test/java/org/minions/devfund/roy/WhackAMoleTest.java

```diff
@@ -0,0 +1,201 @@
+package org.minions.devfund.roy;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Random;
+
+/**
+ * Test for WhackAMole class.
+ */
+public class WhackAMoleTest {
+    private WhackAMole whackAMole;
+
+    /**
+     * Test negative positions for place method.
+     */
+    @Test
+    public void placeNegativePosition() {
+        final int attempts = 2;
+        final int dimension = 2;
+        final int xNegative = -1;
+        final int yNegative = -3;
+        whackAMole = new WhackAMole(attempts, dimension);
+        Assert.assertFalse(whackAMole.place(xNegative, yNegative));
+    }
+
+    /**
+     * Test place a mole in a invalid position.
+     */
+    @Test
+    public void placeAMoleInAValidPosition() {
+        final int attempts = 20;
+        final int dimension = 10;
+
+        whackAMole = new WhackAMole(attempts, dimension);
+        boolean result;
+
+        for (int i = 0; i < dimension; i++) {
+            for (int j = 0; j < dimension; j++) {
+                result = whackAMole.place(i, j);
+                if (result) {
+                    Assert.assertTrue(result);
+                    break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Test place method for an invalid position.
+     */
+    @Test
+    public void placeAMoleInAnInvalidPosition() {
+        final int attempts = 30;
+        final int dimension = 10;
+
+        whackAMole = new WhackAMole(attempts, dimension);
+        boolean result = true;
+
+        for (int i = 0; i < dimension; i++) {
+            for (int j = 0; j < dimension; j++) {
+                result = whackAMole.place(i, j);
+                if (!result) {
+                    Assert.assertFalse(result);
+                    break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Tests whack method.
+     */
+    @Test
+    public void whack() {
```

Comment:
Since your pull request is still open.
please write your unit tests something like this.

Running: testConstructorGrid
Finished: testConstructorGrid
Running: testConstructorNumOfAttempts
Finished: testConstructorNumOfAttempts
Running: testPlaceFail
Finished: testPlaceFail
Running: testWhackSucceedAttempts
Finished: testWhackSucceedAttempts
Running: testWhackFailAttempts
Finished: testWhackFailAttempts
Running: testPlaceSucceed
Finished: testPlaceSucceed
Running: testWhackFailMolesLeft
Finished: testWhackFailMolesLeft
Running: testWhackSucceedMolesLeft
Finished: testWhackSucceedMolesLeft